src/chainstate/stacks/events.rs

```diff
@@ -279,6 +295,30 @@ impl NFTMintEventData {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct NFTBurnEventData {
+    pub asset_identifier: AssetIdentifier,
+    pub sender: PrincipalData,
+    pub value: Value,
+}
+
+impl NFTBurnEventData {
+    pub fn json_serialize(&self) -> serde_json::Value {
+        let raw_value = {
+            let mut bytes = vec![];
+            self.value.consensus_serialize(&mut bytes).unwrap();
+            let formatted_bytes: Vec<String> = bytes.iter().map(|b| format!("{:02x}", b)).collect();
+            formatted_bytes
+        };
+        json!({
+            "asset_identifier": format!("{}", self.asset_identifier),
+            "sender": format!("{}",self.sender),
```

Comment:
No need to change, but just a note that `.to_string()` is a an auto-implemented function that is more or less identical to `format!("{}", ...)`.